UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API/Services/ProductDataService.cs

```diff
@@ -243,7 +243,11 @@ public async Task<ExchangeSetServiceResponse> CreateProductDataSinceDateTime(Pro
             DateTime salesCatalogueServiceRequestCompletedAt = DateTime.UtcNow;
             monitorHelper.MonitorRequest("Sales Catalogue Service Since DateTime Request", salesCatalogueServiceRequestStartedAt, salesCatalogueServiceRequestCompletedAt, productDataSinceDateTimeRequest.CorrelationId, null, null, fileSize, null);
 
-            IEnumerable<string> aioCells = FilterAioCellsByProductData(salesCatalogueResponse.ResponseBody);
+            // filter cells method might update the number of items in the list so get total count before calling it
+            var allCellsCount = salesCatalogueResponse.ResponseBody?.Products.Count;
```

Comment:
This value (allCellsCount) isn't used anywhere - can it be removed? All looks good otherwise!

Reply:
Done.